typespec-extension/src/code-model-builder.ts

```diff
@@ -1327,6 +1317,9 @@ export class CodeModelBuilder {
   }
 
   private processParameterBody(op: CodeModelOperation, httpOperation: HttpOperation, body: ModelProperty | Model) {
+    // set contentTypes to mediaTypes
+    op.requests![0].protocol.http!.mediaTypes = httpOperation.parameters.body!.contentTypes;
```

Comment:
This `body.contentTypes` was added to http lib a few weeks ago (part of multipart PR).

Given this, we should no longer need to use the `content-type` header.